openwhisk/.gitignore

```diff
@@ -0,0 +1,46 @@
+##############################################
```

Comment:
Any chance you can remove this file ?  See here for the reasoning why gitignore should be kept local https://julien.danjou.info/properly-managing-your-gitignore/

Reply:
@chmouel the design of gitignore supports and encourages use of directory-specific use (see https://labs.consol.de/development/git/2017/02/22/gitignore.html) top allow finer-grained exclusion.  It seems that since the catalog is effectively a collection of subprojects each would want and be encouraged to have its own rules specific to its installation/configuration/tests.  In fact, if you follow the readme's optional instructions to manually runt the examples the scripts and steps cause many new files to be created which we do not want checked int by developers who may be updating this specific catalog entry.  This gitignore reflects those items my fellow developers actually derived from those "oops" moments while working on this specific entry over the last year.

Reply:
I am not so much convinced by this argument of having gitignore everywhere in the catalogs, if the gitignore contained some specific files for the specifics openwhisk files then that would be fine, but those are mostly editors/ide exclude, which are best kept in the user configuration

Reply:
@chmouel the design of gitignore supports and encourages use of directory-specific use (see https://labs.consol.de/development/git/2017/02/22/gitignore.html) to allow finer-grained exclusion. It seems that since the catalog is effectively a collection of subprojects each would want and be encouraged to have its own rules specific to its installation/configuration/tests.

In fact, if you follow the "openwhisk" specific readme's optional instructions to manually run the examples the scripts and steps cause many new files to be created which we do not want checked int by developers who may be updating this specific catalog entry. This gitignore reflects those items my fellow developers actually derived from those "oops" moments while working on this specific entry over the last year.

However, I agree with the point from the ref. article and removed the "editor" specific entries in the file.  Hopefully, what remains is accurate to the examples only.

Reply:
> I am not so much convinced by this argument of having gitignore everywhere in the catalogs, if @chmouel As indicated on the update, gitignore is specific to the openwhisk entry and artifacts that are created from following its README only.

Reply:
@chmouel I decided to co-locate the JSON parameter (input) files in the same subdirectory with the Java pipelineRun YAML as this is the only pipelinerun (example) that uses these input files.